connectors-e2e-test/connectors-e2e-test-agentic-ai/src/test/java/io/camunda/connector/e2e/agenticai/AiAgentTests.java

```diff
@@ -286,13 +324,259 @@ void executesAgentWithToolCallingAndUserFeedback() throws Exception {
 
       assertThat(jobWorkerCounter.get()).isEqualTo(2);
     }
+
+    @ParameterizedTest
+    @CsvSource({
+      "test.csv,base64,text/csv",
+      "test.gif,base64,image/gif",
+      "test.jpg,base64,image/jpeg",
+      "test.json,base64,application/json",
+      "test.pdf,base64,application/pdf",
+      "test.png,base64,image/png",
+      "test.txt,text,text/plain",
+      "test.webp,base64,image/webp",
+      "test.xml,base64,application/xml",
+      "test.yaml,base64,application/yaml"
+    })
+    void supportsDocumentResponsesFromToolCalls(String filename, String type, String mimeType)
+        throws Exception {
+      String expectedData;
+      if (type.equals("text")) {
+        expectedData = testFileContent(filename).get().replaceAll("\\n", "\\\\n");
+      } else {
+        expectedData = testFileContentBase64(filename).get();
+      }
+
+      final var initialUserPrompt = "Go and download a document!";
+      final var expectedConversation =
+          List.of(
+              new SystemMessage(
+                  "You are a helpful AI assistant. Answer all the questions, but always be nice. Explain your thinking."),
+              new UserMessage(initialUserPrompt),
+              new AiMessage(
+                  "The user asked me to download a document. I will call the Download_A_File tool to do so.",
```

Comment:
`Download_A_File` is a REST connector using the "Store Response" option.

![image](https://github.com/user-attachments/assets/402c49d9-6c65-4295-9a52-95d1ab4a14b8)